File: kojt/src/kadai1-1.py
import math
import random

F_SAMP = 40000


def create_sin(n, a, f, p):
    ans = []
    for i in range(n):
        ans.append([i/F_SAMP, a*math.sin((2*math.pi*f*i)/F_SAMP)])
    return ans


def create_bpsk(n, rate):
    ans = []
    for i in range(n):
        buf = random.randrange(2)
        for r in range(rate):
            ans.append([(i*rate+r)/F_SAMP, buf])
    return ans


def create_ask(cs, bs):
    ans = []
    for c, b in zip(cs, bs):
        ans.append([c[0], c[1]*b[1]])
    return ans


if __name__=='__main__':
    c = create_sin(1200, 1, 40000, -math.pi/2)
    b = create_bpsk(6, 200)
    datas = create_ask(c, b)
    with open('../txt/kadai1-3.txt', 'w') as f:
        for data in datas:
            f.write(str(data[0]) + '  ' + str(data[1])+'\n')

File: kojt/src/kadai1-1.c
#include <stdio.h>
#include <math.h>

#define F_SAMP 200

int main(){

    return 0;
}

double create_sin(int n, double a, double f, double p){
    double *ans;
    ans = (double *)malloc(sizeof(double)*n);
    for(i=0; i<n; i++){
        ans[i] = a*sin((2*PI*f*n)/(F_SAMP))
    }

}